src/data_source/storage/sql.rs

```diff
@@ -235,6 +236,8 @@ impl Default for Config {
 impl Default for Config {
     fn default() -> Self {
         SqliteConnectOptions::default()
+            .journal_mode(sqlx::sqlite::SqliteJournalMode::Wal)
+            .synchronous(sqlx::sqlite::SqliteSynchronous::Normal)
```

Comment:
I'm wondering about this. Based on [the docs](https://www.sqlite.org/pragma.html#pragma_synchronous), this is almost always safe with WAL mode, _however_, the host restarting at just the wrong time could very rarely cause us to lose committed data. Since we base our votes on data being successfully committed, this weakens the guarantee that, e.g., at least one honest node has the leaf corresponding to each QC. Since AWS does have the propensity to restart services without warning, this worries me a bit.

It is probably an extremely low-probability failure, but if it is not prohibitively expensive to run in `FULL` mode, I think we should just do that and be extra robust